packages/interactivity/src/utils.ts

```diff
@@ -374,3 +374,23 @@ export const isPlainObject = (
 			typeof candidate === 'object' &&
 			candidate.constructor === Object
 	);
+
+/**
+ * Indicates that the passed `callback` requires synchronous access to the event object.
+ *
+ * @param callback The event callback.
+ * @return Wrapped event callback.
+ */
+export const withSyncEvent = ( callback: Function ): Function => {
+	if ( callback?.constructor?.name === 'GeneratorFunction' ) {
+		const wrapped = function* ( ...args: any[] ) {
+			yield* callback( ...args );
+		};
+		wrapped.sync = true;
+		return wrapped;
+	}
+
+	const wrapped = ( ...args: any[] ) => callback( ...args );
```

Comment:
Should this be preserving `this` context like you're doing in `wrapEventAsync`?

```suggestion
	const wrapped = function ( ...args: any[] ) {
		return callback.apply( this, args )
	};
```

Reply:
Same for the generator variant above.

Reply:
Done as part of https://github.com/WordPress/gutenberg/commit/bf3dbcc861367a38131ac0943ad49f60e2474912